### PR TITLE
ENH: Add support for bounds class in curve_fit

### PR DESCRIPTION
**Pull request by dschmitz89**
#### What does this implement/fix?
Recently, support for the `Bounds` class was added to `least_squares`. I think it makes sense to support it also in  `curve_fit` .

So far, the bounds for `curve_fit` have to be provided in a different way than for `minimize` and the global optimizers. This complicates the overall API of `scipy.optimize`. Now, almost all optimizers support the `Bounds` class.

### CODE REVIEW

File: scipy/optimize/_minpack_py.py
Comment (tupui) [inline scipy/optimize/_minpack_py.py:606]: At that point I would remove this or we should add a `versionchanged` to the other bullet if we want to be extra correct.

**Review by tupui — APPROVED** [review]
Thanks @dschmitz89. LGTM, but I  would like a second pair of eyes. If nothing happens in 2 weeks, I will merge.

**Review by mdhaber — APPROVED** [review]
Seems pretty straightforward. Documentation renders correctly, and the test is as good as the original (although it might be nice to see that the minimum w/out bounds if at [2, 2] so that the test really shows that bounds is doing something).

File: scipy/optimize/_minpack_py.py
Comment (dschmitz89) [reply scipy/optimize/_minpack_py.py:606]: I also think the `curve_fit` docs are overloaded with "New in version x" boxes. But will let maintainers decide, seems like a policy issue.

File: scipy/optimize/_minpack_py.py
Comment (tupui) [inline scipy/optimize/_minpack_py.py:607]:
```suggestion
              bounds on all or some parameters.

    method : {'lm', 'trf', 'dogbox'}, optional
```

**Pull request merged**
